src/urllib3/contrib/http2.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import annotations
+
+import contextlib
+import threading
+import typing
+
+import h2.config  # type: ignore[import]
+import h2.connection  # type: ignore[import]
+import h2.events  # type: ignore[import]
+
+import urllib3.connection
+import urllib3.util.ssl_
+
+from .._collections import HTTPHeaderDict
+from ..connection import HTTPSConnection
+from ..connectionpool import HTTPSConnectionPool
+
+orig_HTTPSConnection = HTTPSConnection
+
+
+class HTTP2Connection(HTTPSConnection):
```

Comment:
I opened an issue for this feature: https://github.com/urllib3/urllib3/issues/3303

I think for now we should focus on HTTPS, but it'll be good to have this for the first release since it may require some rejiggering of the "HTTP/2 startup"? Moves the figuring out HTTP version stage from being only on handshake to being later in an actual HTTP transaction.